### ISSUE TITLE
missing translation passes MissingTranslation instead MissingTranslationData

### ISSUE THREAD

**Issue by pivotalhalogen** [svenfuchs/i18n#134]
Given this (a more specific version of the suggestion from rails guides http://guides.rubyonrails.org/i18n.html#using-different-exception-handlers ):

``` ruby

module I18n
  def self.just_raise_that_exception(exception, locale, key, options)
    raise exception
  end
end

I18n.exception_handler = :just_raise_that_exception
```

and this test:

``` ruby
require 'spec_helper'

describe "MissingTranslationData" do
  it "should raise an error for missing translation" do
    expect {
      I18n.t('totally.bogus.translation.key')
    }.to raise_exception(I18n::MissingTranslationData)
  end
end
```

You get the following failure:

```
expected I18n::MissingTranslationData, got #<TypeError: exception class/object expected>
```

Because `exception` is `I18n::MissingTranslation` not `I18n::MissingTranslationData` and `I18n::MissingTranslation` does not inherit from an Exception class.

**Comment by radar** [comment]
Thanks @pivotalhalogen. I can confirm that I can reproduce this issue. Taking a look.

**Comment by radar** [comment]
I have a fix for this in 1bcc77b. Thanks @pivotalhalogen and @henrik!
